google/cloud/aiplatform/_featurestores/entity_type.py

```diff
@@ -0,0 +1,456 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import datetime
+import pandas as pd
+
+from google.auth import credentials as auth_credentials
+from google.protobuf import field_mask_pb2
+
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform.compat.types import entity_type as gca_entity_type
+from google.cloud.aiplatform import _featurestores
+from google.cloud.aiplatform import utils
+
+from google.cloud.aiplatform.utils import featurestore_utils
+
+_LOGGER = base.Logger(__name__)
+_ALL_FEATURE_IDS = "*"
+
+
+class EntityType(base.VertexAiResourceNounWithFutureManager):
+    """Managed entityType resource for Vertex AI."""
+
+    client_class = utils.FeaturestoreClientWithOverride
+
+    _is_client_prediction_client = False
+    _resource_noun = None
+    _getter_method = "get_entity_type"
+    _list_method = "list_entity_types"
+    _delete_method = "delete_entity_type"
+
+    def __init__(
+        self,
+        entity_type_name: str,
```

Comment:
Is this a common pattern in other resource? I personally find this a bit confuse for overload this arg with both short form and long form, which make a weird case when full path is used, but project/location/featurestore_id also provided. Is this possible to have overloading constructor in Python?

Reply:
Python does not support overloading constructor, that's why we have this as the common pattern for all other Vertex SDK resource classes, for example this is Vertex AI SDK [Dataset class constructor](https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/datasets/dataset.py#L61).

Agree on the confusion of the usage of optional arguments, but that's the language limitation we are working with. We try to compensate by emphasizing the Docstring and Example Usage throughout the SDK.